src/StripeTests/Services/Coupons/StripeCouponServiceTest.cs

```diff
@@ -0,0 +1,112 @@
+namespace StripeTests
+{
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+
+    using Stripe;
+    using Xunit;
+
+    public class StripeCouponServiceTest : BaseStripeTest
+    {
+        private const string CouponId = "co_123";
+
+        private StripeCouponService service;
+        private StripeCouponCreateOptions createOptions;
+        private StripeCouponUpdateOptions updateOptions;
+        private StripeCouponListOptions listOptions;
+
+        public StripeCouponServiceTest()
+        {
+            this.service = new StripeCouponService();
+
+            this.createOptions = new StripeCouponCreateOptions()
+            {
+                PercentOff = 25,
+                Duration = "forever",
+            };
+
+            this.updateOptions = new StripeCouponUpdateOptions()
+            {
+                Metadata = new Dictionary<string, string>()
+                {
+                    { "key", "value" },
+                },
+            };
+
+            this.listOptions = new StripeCouponListOptions()
+            {
+                Limit = 1,
```

Comment:
Do we care to set that `Limit` if we don't explicitly test it?

Reply:
Yeah, I think so -- just like for creation and update requests, it at least ensures that the method does accept a `StripeCouponListOptions` and that the resulting encoding is accepted by stripe-mock.

Reply:
sure but I meant you could do this without setting `Limit`.